static/app/bootstrap/bootstrapRequests.tsx

```diff
@@ -0,0 +1,200 @@
+    gcTime: BOOTSTRAP_QUERY_GC_TIME,
+    retry: false,
+  });
+}
+
+/**
+ * The TeamsStore expects a cursor, hasMore, and teams
+ * Since some of this information exists in headers, parse it into something we can serialize
+ */
+function createTeamsObject(response: ApiResult): {
+  cursor: string | null;
+  hasMore: boolean;
+  teams: Team[];
+} {
+  const teams = response[0];
+  const paginationObject = parseLinkHeader(response[2]!.getResponseHeader('Link'));
+  const hasMore = paginationObject?.next?.results ?? false;
+  const cursor = paginationObject.next?.cursor ?? null;
+  return {teams, hasMore, cursor};
+}
+
+export function getBoostrapTeamsQueryOptions(orgSlug: string | null) {
+  return queryOptions({
+    queryKey: ['bootstrap-teams', orgSlug],
+    queryFn: async (): Promise<{
+      cursor: string | null;
+      hasMore: boolean;
+      teams: Team[];
+    }> => {
+      const uncancelableApi = new Client();
+      // Get the preloaded data promise
+      try {
+        const preloadResponse = await getPreloadedData('teams', orgSlug!);
+        // If the preload request was successful, find the matching team
+        if (preloadResponse !== null && preloadResponse[0] !== null) {
+          return createTeamsObject(preloadResponse);
+        }
+      } catch {
+        //
+      }
+
+      const teamsApiResponse = await uncancelableApi.requestPromise(
+        `/organizations/${orgSlug}/teams/`,
+        {
+          includeAllArgs: true,
+        }
+      );
+      return createTeamsObject(teamsApiResponse);
+    },
+    enabled: !!orgSlug,
+    staleTime: BOOTSTRAP_QUERY_STALE_TIME,
+    gcTime: BOOTSTRAP_QUERY_GC_TIME,
+    retry: false,
+  });
+}
+
+export function getBootstrapProjectsQueryOptions(orgSlug: string | null) {
+  return queryOptions({
+    queryKey: ['bootstrap-projects', orgSlug],
+    queryFn: async (): Promise<Project[]> => {
+      const uncancelableApi = new Client();
+      // Get the preloaded data promise
+      try {
+        const preloadResponse = await getPreloadedData('projects', orgSlug!);
+        // If the preload request was successful
+        if (preloadResponse !== null && preloadResponse[0] !== null) {
+          return preloadResponse[0];
+        }
+      } catch {
+        //
+      }
+
+      const [projects] = await uncancelableApi.requestPromise(
+        `/organizations/${orgSlug}/projects/`,
+        {
+          includeAllArgs: true,
+          query: {
+            all_projects: 1,
+            collapse: ['latestDeploys', 'unusedFeatures'],
+          },
+        }
+      );
+      return projects;
+    },
+    enabled: !!orgSlug,
+    staleTime: BOOTSTRAP_QUERY_STALE_TIME,
+    gcTime: BOOTSTRAP_QUERY_GC_TIME,
+    retry: false,
+  });
+}
+
+/**
+ * Small helper to access the preload requests in window.__sentry_preload
+ */
+function getPreloadedData(
+  name: 'organization' | 'projects' | 'teams',
+  slug: string
+): Promise<ApiResult | null> {
```

Comment:
I think we could also make this function synchronous, as there isn’t really anything async going on here 🤔